### DISCUSSION TITLE
CodeQL ASPNETCOMPILER error 1003

### DISCUSSION THREAD

**Discussion by sc00terUL** [community/community#68250] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello,

I am attempting to get CodeQL to work for our solution.  I can't get past the "build app for release" step.  I keep getting the error in the subject

Where does CodeQL get the directory to drop the output files to?  From the .sln?  from the .csproj?

We are building a .NET framework 4.8 web app with supporting libraries and nuget packages.  All the nuget gets pulled in and our supporting library files get built correctly, but when attempting to copy all the supporting dlls into the "output" directory, it complains that the directory doesn't exist.

**Comment by Neved4** [comment, marked as answer]
@sc00terUL Typically, CodeQL will be using here the output dir from your project settings, specifically from the `.csproj` file. Is the `<OutputPath>` inside `.csproj` looking good?

**Comment by greatlarrymartin** [comment]
When using CodeQL, the output directory for the build files is determined by the build process of your project, which is typically specified in the project file (.csproj) rather than the solution file (.sln).

To resolve the error you are encountering, follow these steps:

1. Open your .csproj file in a text editor or an integrated development environment (IDE) such as Visual Studio.
2. Look for the `<OutputPath>` element within the `<PropertyGroup>` section of the .csproj file. This element specifies the directory where the build output will be placed. It should look something like this:

```xml
<OutputPath>bin\Debug\</OutputPath>
```

3. Ensure that the specified output directory exists in your project structure. If it doesn't, create the directory manually.

4. Save the .csproj file and try building the project again. CodeQL should now be able to copy the supporting DLLs into the specified output directory without any complaints.

Remember to adjust the configuration (e.g., "Debug" or "Release") in the `<OutputPath>` element if you are building for a different configuration.

If you continue to experience issues or need further assistance, please provide more details about the steps you have taken and any specific error messages you receive. This will help the community provide more targeted suggestions to help you resolve the problem.

**Reply by Neved4** [reply]
I do love ChatGPT
